### PR TITLE
Add feature repository and types

### PR DESCRIPTION
**Pull request by N-i-z**

### CODE REVIEW

File: apps/webapp/src/lib/server/features/features.types.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/**
+ * Features.
+ */
+export interface Features {
```
Comment (thani-sh) [inline apps/webapp/src/lib/server/features/features.types.ts]: Use singular form unless its an array

```suggestion
export interface Feature {
```

File: apps/webapp/src/lib/server/features/features.repository.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+export async function findById(id: string): Promise<Features | null> {
+    const coll = await getCollection();
+    const doc = await coll.findOne({ _id: new ObjectId(id) });
+    return doc ? fromDocument(doc) : null;
+}
+
+/**
+ * Find a feature by name.
+ */
+export async function findByName(name: string): Promise<Features | null> {
```
Comment (thani-sh) [inline apps/webapp/src/lib/server/features/features.repository.ts]: We won't need this function. I'll explain later.

File: apps/webapp/src/lib/server/features/features.repository.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+ * The name of the MongoDB collection for features.
+ */
+const COLLECTION_NAME = 'features';
+
+/**
+ * Type for the MongoDB document for Features in db.
+ */
+type FeatureDocument = {
+    _id: ObjectId;
+    name: string;
```
Comment (thani-sh) [inline apps/webapp/src/lib/server/features/features.repository.ts]: Can you add a `key` field here and a `findByKey` function.

We can also add an `organizationId` field here so we can find out to which organization this feature belongs to. It'll also help to have a function to get all features that belong to an organization ( `findByOrganization` perhaps ).

You can check the `findByMember` function on organization repository file for an example where we query and fetch multiple documents.

**Review by thani-sh — CHANGES_REQUESTED** [review]

File: apps/webapp/src/lib/server/features/features.types.ts (hunk 1 above)
Comment (N-i-z) [reply apps/webapp/src/lib/server/features/features.types.ts]: oki

File: apps/webapp/src/lib/server/features/features.repository.ts (hunk 2 above)
Comment (N-i-z) [reply apps/webapp/src/lib/server/features/features.repository.ts]: oki
as i said i just copy pasted the existing code and tweaked it a bit

File: apps/webapp/src/lib/server/features/feature.repository.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
+import { getMongoClient } from '$lib/server/database';
+import { ObjectId, type Collection } from 'mongodb';
+import type { Feature } from './feature.types';
+
+/**
+ * The name of the MongoDB collection for features.
+ */
+const COLLECTION_NAME = 'feature';
```
Comment (thani-sh) [inline apps/webapp/src/lib/server/features/feature.repository.ts]: Use plural form

```suggestion
const COLLECTION_NAME = 'features';
```

File: apps/webapp/src/lib/server/features/feature.repository.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+/**
+ * Type for the MongoDB document for Features in db.
+ */
+type FeatureDocument = {
+    _id: ObjectId;
+    name: string;
+    key: string;
+    organization: string[];
```
Comment (thani-sh) [inline apps/webapp/src/lib/server/features/feature.repository.ts]: Features (also projects and feedback) can only belong to one organization. Also it's more clear if we say that it's an id.

```suggestion
    organizationId: string;
```

File: apps/webapp/src/lib/server/features/feature.repository.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+export async function findByKey(key: string): Promise<Feature | null> {
+    const coll = await getCollection();
+    const doc = await coll.findOne({ key: new String(key) });
+    return doc ? fromDocument(doc) : null;
+}
+
+/**
+ * Find a feature by organisation
+ */
+export async function findByOrganisation(OrganizationId: string): Promise<Feature[]> {
```
Comment (thani-sh) [inline apps/webapp/src/lib/server/features/feature.repository.ts]: Use lower camelCase for variable names.

```suggestion
export async function findByOrganisation(organizationId: string): Promise<Feature[]> {
```

**Review by thani-sh — CHANGES_REQUESTED** [review]

File: apps/webapp/src/lib/server/features/feature.repository.ts [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+const COLLECTION_NAME = 'features';
+
+/**
+ * Type for the MongoDB document for Features in db.
+ */
+type FeatureDocument = {
+    _id: ObjectId;
+    name: string;
+    key: string;
+    organizationId: string;
```
Comment (thani-sh) [inline apps/webapp/src/lib/server/features/feature.repository.ts]: Shall we change to ObjectId? Will be useful for some database queries

File: apps/webapp/src/lib/server/features/feature.repository.ts [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+    const doc = await coll.findOne({ _id: new ObjectId(id) });
+    return doc ? fromDocument(doc) : null;
+}
+
+/**
+ * Find a feature by key.
+ */
+export async function findByKey(key: string): Promise<Feature | null> {
+    const coll = await getCollection();
+    const doc = await coll.findOne({ key: new String(key) });
```
Comment (thani-sh) [inline apps/webapp/src/lib/server/features/feature.repository.ts]: No need to do `new String` here because it's already a string.

**Review by thani-sh — CHANGES_REQUESTED** [review]

File: apps/webapp/src/lib/server/features/feature.repository.ts (hunk 7 above)
Comment (N-i-z) [reply apps/webapp/src/lib/server/features/feature.repository.ts]: umm change what to ObjectId?

File: apps/webapp/src/lib/server/features/feature.repository.ts [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+export async function findByKey(key: string): Promise<Feature | null> {
+    const coll = await getCollection();
+    const doc = await coll.findOne({ key });
+    return doc ? fromDocument(doc) : null;
+}
+
+/**
+ * Find a feature by organisation
+ */
+export async function findByOrganisation(organizationId: ObjectId): Promise<Feature[]> {
```
Comment (thani-sh) [inline apps/webapp/src/lib/server/features/feature.repository.ts]: We should a string `organizationId` as a parameter and convert it to an ObjectId inside the function. ( check `findById` for an example ).

Comment (N-i-z) [reply apps/webapp/src/lib/server/features/feature.repository.ts]: this is confusing me too

**Review by thani-sh — APPROVED** [review]

**Pull request merged**